pkg/kubelet/eviction/helpers.go

```diff
@@ -444,6 +444,52 @@ func podMemoryUsage(podStats statsapi.PodStats) (v1.ResourceList, error) {
 	}, nil
 }
 
+// localEphemeralVolumeNames returns the set of ephemeral volumes for the pod that are local
+func localEphemeralVolumeNames(pod *v1.Pod) []string {
+	result := []string{}
+	for _, volume := range pod.Spec.Volumes {
+		if volume.GitRepo != nil ||
+			(volume.EmptyDir != nil && volume.EmptyDir.Medium != v1.StorageMediumMemory) ||
+			volume.ConfigMap != nil || volume.Secret != nil ||
```

Comment:
Secret is of medium memory. May be we can ignore it?

Reply:
Found that ` secret volumes are backed by tmpfs (a RAM-backed filesystem) so they are never written to non-volatile storage` in https://kubernetes.io/docs/concepts/storage/volumes/#secret

Reply:
Will modify soon and also add some e2e test cases